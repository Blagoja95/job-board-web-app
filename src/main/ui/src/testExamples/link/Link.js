import {useState} from "react";

const Link = ({page, childredn}) => {
	const state = {
		HOVERED: 'hovered',
		NORMAL: 'normal'
	};

	const [status, setStatus] = useState(state.NORMAL);

	const onMouseEnter = () => {
		setStatus(state.HOVERED)
	};
	const onMouseLeave = () => {
		setStatus(state.NORMAL)
	};

	return <a className={status}
			  href={page || '#'}
			  onMouseEnter={onMouseEnter}
			  onMouseLeave={onMouseLeave}
	> {childredn}
	</a>
};

export default Link;