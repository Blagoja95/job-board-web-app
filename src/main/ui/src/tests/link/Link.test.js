import renderer from 'react-test-renderer';
import Link from '../../testExamples/link/Link';

describe("TestExamples", () => {
	describe("Link", () => {
		it('Class is changed when hovered', () => {
			const cmp = renderer.create(
				<Link page="http://www.facebook.com">Facebook</Link>,
			);

			let tree = cmp.toJSON();
			expect(tree).toMatchSnapshot();

			renderer.act(() => {
				tree.props.onMouseEnter();
			});

			tree = cmp.toJSON();
			expect(tree).toMatchSnapshot();

			renderer.act(() => {
				tree.props.onMouseLeave();
			});

			tree = cmp.toJSON();
			expect(tree).toMatchSnapshot();
		});
	});
});
