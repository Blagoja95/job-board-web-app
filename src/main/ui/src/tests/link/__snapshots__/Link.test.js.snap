// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestExamples Link Class is changed when hovered 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
   
</a>
`;

exports[`TestExamples Link Class is changed when hovered 2`] = `
<a
  className="hovered"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
   
</a>
`;

exports[`TestExamples Link Class is changed when hovered 3`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
   
</a>
`;
